execution/executor-service/src/remote_state_view.rs

```diff
@@ -114,25 +123,61 @@ impl RemoteStateViewClient {
 
     pub fn init_for_block(&self, state_keys: Vec<StateKey>) {
         *self.state_view.write().unwrap() = RemoteStateView::new();
-        self.pre_fetch_state_values(state_keys);
+        info!(
+            "REMOTE_EXECUTOR: Prefetching {} state keys",
+            state_keys.len()
+        );
+        self.pre_fetch_state_values(state_keys, false);
     }
 
-    fn pre_fetch_state_values(&self, state_keys: Vec<StateKey>) {
+    fn insert_keys_and_fetch_values(
+        state_view_clone: Arc<RwLock<RemoteStateView>>,
+        thread_pool: Arc<ThreadPool>,
```

Comment:
Doesn't allow me to take reference because I am spawning a new thread later. Below is the error:

> 166 |     fn pre_fetch_state_values(&self, state_keys: Vec<StateKey>, sync_insert_keys: bool) {
    |                               -----
    |                               |
    |                               `self` is a reference that is only valid in the method body
    |                               let's call the lifetime of this reference `'1`
...
189 |             self.thread_pool.spawn(insert_and_fetch);
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             `self` escapes the method body here
    |             argument requires that `'1` must outlive `'static`